src/test/java/com/artipie/docker/DockerProxyIT.java

```diff
@@ -100,4 +97,60 @@ void shouldPullRemote() throws Exception {
             "docker", "pull", img
         );
     }
+
+    @Test
+    void shouldPushAndPull() throws Exception {
+        final String image = "artipie:8080/my-docker/alpine:3.11";
+        this.deployment.assertExec(
+            "Failed to login to Artipie",
+            new ContainerResultMatcher(),
+            "docker", "login",
+            "--username", "alice",
+            "--password", "123",
+            "artipie:8080"
+        );
+        this.deployment.assertExec(
+            "Failed to pull origin image",
+            new ContainerResultMatcher(
+                new IsEqual<>(ContainerResultMatcher.SUCCESS),
+                new StringContains(
+                    "Status: Downloaded newer image for alpine:3.11"
+                )
+            ),
+            "docker", "pull", "alpine:3.11"
+        );
+        this.deployment.assertExec(
+            "Failed to tag origin image",
+            new ContainerResultMatcher(),
+            "docker", "tag", "alpine:3.11", image
+        );
+        this.deployment.assertExec(
+            "Failed to push image to Artipie",
+            new ContainerResultMatcher(
+                new IsEqual<>(ContainerResultMatcher.SUCCESS),
+                new StringContains(
+                    "The push refers to repository [artipie:8080/my-docker/alpine]"
+                )
+            ),
+            "docker", "push", image
+        );
```

Comment:
Why does it work? You are pushing image to proxy registry with `registry-1.docker.io` and `mcr.microsoft.com` remotes, but the proxy doesn't have push permissions in these remotes, isn't it?

Reply:
Sure it doesn't have; it pushes an image to the local storage as declared in the [doc](https://github.com/artipie/artipie/tree/master/examples/docker#docker-proxy-repo).
The task description is 
`Docker proxy supports pushing to local storage if such storage is specified. 
 It should be verified that an image can be pushed to a proxy repository and pulled later`.
It looks like we should test `pushing to local storage,` so I think all is ok.